kroxylicious/src/test/java/io/kroxylicious/proxy/internal/filter/FetchResponseTransformationFilterTest.java

```diff
@@ -0,0 +1,219 @@
+    private ArgumentCaptor<Integer> bufferInitialCapacity;
+
+    @Captor
+    private ArgumentCaptor<ResponseHeaderData> responseHeaderDataCaptor;
+
+    @Captor
+    private ArgumentCaptor<ApiMessage> apiMessageCaptor;
+
+    @BeforeEach
+    @SuppressWarnings("unchecked")
+    void setUp() {
+        filter = new FetchResponseTransformationFilter(new FetchResponseTransformationConfig(ProduceRequestTransformationFilter.UpperCasing.class.getName()));
+
+        when(context.forwardResponse(responseHeaderDataCaptor.capture(), apiMessageCaptor.capture())).thenAnswer(
+                invocation -> CompletableFuture.completedStage(responseFilterResult));
+        when(context.responseFilterResultBuilder()).thenReturn(responseFilterResultBuilder);
+
+        when(responseFilterResult.message()).thenAnswer(invocation -> apiMessageCaptor.getValue());
+        when(responseFilterResult.header()).thenAnswer(invocation -> responseHeaderDataCaptor.getValue());
+
+        when(responseFilterResultBuilder.forward(responseHeaderDataCaptor.capture(), apiMessageCaptor.capture()))
+                .thenReturn((CloseOrTerminalStage<ResponseFilterResult>) responseFilterResultBuilder);
+        when(((CloseOrTerminalStage<ResponseFilterResult>) responseFilterResultBuilder).completed()).thenReturn(CompletableFuture.completedStage(responseFilterResult));
+
+        when(context.createByteBufferOutputStream(bufferInitialCapacity.capture())).thenAnswer(
+                (Answer<ByteBufferOutputStream>) invocation -> {
+                    Object[] args = invocation.getArguments();
+                    Integer size = (Integer) args[0];
+                    return new ByteBufferOutputStream(size);
+                });
+    }
+
+    @Test
+    void filterHandlesResponseBasedOnTopicNames() throws Exception {
+
+        var fetchResponse = new FetchResponseData();
+        fetchResponse.responses().add(createFetchableTopicResponseWithOneRecord(RECORD_KEY, ORIGINAL_RECORD_VALUE).setTopic(TOPIC_NAME)); // Version 12
+
+        var stage = filter.onFetchResponse(fetchResponse.apiKey(), new ResponseHeaderData(), fetchResponse, context);
+        assertThat(stage).isCompleted();
+
+        var filteredResponse = (FetchResponseData) stage.toCompletableFuture().get().message();
+        var filteredRecords = responseToRecordStream(filteredResponse).toList();
+        assertThat(filteredRecords)
+                .withFailMessage("unexpected number of records in the filter response")
+                .hasSize(1);
+
+        var filteredRecord = filteredRecords.get(0);
+        assertThat(decodeUtf8Value(filteredRecord))
+                .withFailMessage("expected record value to have been transformed")
+                .isEqualTo(EXPECTED_TRANSFORMED_RECORD_VALUE);
+    }
+
+    @Test
+    void filterHandlesResponseBasedOnTopicIds() throws Exception {
+
+        var fetchResponse = new FetchResponseData();
+        // Version 13 switched to topic id rather than topic names.
```

Comment:
nit: rather than including the version as a comment should we include it in the test name?